pkg/controller/containerrecreaterequest/crr_controller.go

```diff
@@ -91,13 +91,14 @@ func add(mgr manager.Manager, r *ReconcileContainerRecreateRequest) error {
 	}
 
 	// Watch for changes to ContainerRecreateRequest
-	err = c.Watch(source.Kind(mgr.GetCache(), &appsv1alpha1.ContainerRecreateRequest{}), &handler.EnqueueRequestForObject{})
+	k := source.Kind(mgr.GetCache(), &appsv1alpha1.ContainerRecreateRequest{}, &handler.TypedEnqueueRequestForObject[*appsv1alpha1.ContainerRecreateRequest]{})
+	err = c.Watch(k)
 	if err != nil {
 		return err
 	}
 
 	// Watch for pod for jobs that have pod selector
-	err = c.Watch(source.Kind(mgr.GetCache(), &v1.Pod{}), &podEventHandler{Reader: mgr.GetCache()})
+	err = c.Watch(source.Kind(mgr.GetCache(), &v1.Pod{}, &podEventHandler{Reader: mgr.GetCache()}))
```

Comment:
why not wrap v1.Pod as client.Object here ?

Reply:
My strategy is to use "typed" if we can clearly know the specific type of "watch." The exceptions are scenarios involving "WS," "UD," or other cases where CRDs  are being watched.